include/aspect/vofinterface/vof_system.h

```diff
@@ -0,0 +1,148 @@
+/*
+ Copyright (C) 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__vofinterface_vof_system_h
+#define __aspect__vofinterface_vof_system_h
+
+#include <aspect/global.h>
+#include <aspect/simulator_access.h>
+
+#include <deal.II/fe/fe_values.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  template <int dim>
+  class Simulator;
+
+  struct AdvectionField;
+
+  namespace internal
+  {
+    namespace Assembly
```

Comment:
I suspect that you only need these things in the VoF method. Maybe move all of it into a namespace `internal::VoF`?